fastly/block_fastly_service_logging_newrelicotlp.go

```diff
@@ -0,0 +1,264 @@
+	return ToServiceAttributeDefinition(&NewRelicOTLPServiceAttributeHandler{
+		&DefaultServiceAttributeHandler{
+			key:             "logging_newrelicotlp",
+			serviceMetadata: sa,
+		},
+	})
+}
+
+// Key returns the resource key.
+func (h *NewRelicOTLPServiceAttributeHandler) Key() string {
+	return h.key
+}
+
+// GetSchema returns the resource schema.
+func (h *NewRelicOTLPServiceAttributeHandler) GetSchema() *schema.Schema {
+	blockAttributes := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The unique name of the New Relic OTLP logging endpoint. It is important to note that changing this attribute will delete and recreate the resource",
+		},
+		"region": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "US",
+			Description: "The region that log data will be sent to. Default: `US`",
+		},
+		"token": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Sensitive:   true,
+			Description: "The Insert API key from the Account page of your New Relic account",
+		},
+	}
+
+	if h.GetServiceMetadata().serviceType == ServiceTypeVCL {
+		blockAttributes["format"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Apache style log formatting. Your log must produce valid JSON that New Relic OTLP can ingest.",
+		}
+		blockAttributes["format_version"] = &schema.Schema{
+			Type:             schema.TypeInt,
+			Optional:         true,
+			Default:          2,
+			Description:      "The version of the custom logging format used for the configured endpoint. Can be either `1` or `2`. (default: `2`).",
+			ValidateDiagFunc: validateLoggingFormatVersion(),
+		}
+		blockAttributes["placement"] = &schema.Schema{
+			Type:             schema.TypeString,
+			Optional:         true,
+			Description:      "Where in the generated VCL the logging call should be placed.",
+			ValidateDiagFunc: validateLoggingPlacement(),
+		}
+		blockAttributes["response_condition"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The name of the condition to apply.",
+		}
+		blockAttributes["url"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The optional New Relic Trace Observer URL to stream logs to for Infinite Tracing.",
+		}
+	}
+
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: blockAttributes,
+		},
+	}
+}
+
+// Create creates the resource.
+func (h *NewRelicOTLPServiceAttributeHandler) Create(_ context.Context, d *schema.ResourceData, resource map[string]any, serviceVersion int, conn *gofastly.Client) error {
+	opts := h.buildCreate(resource, d.Id(), serviceVersion)
+
+	log.Printf("[DEBUG] Fastly New Relic OTLP logging addition opts: %#v", opts)
+
+	return createNewRelicOTLP(conn, opts)
```

Comment:
This method is only one line long so we should just inline its implementation here.

Let's also inline `h.buildCreate()` because the redirection is a cognitive flow disruption (IMHO).